src/monitoring.py

```
@@ -0,0 +1,137 @@
+        try:
+            result = await func(*args, **kwargs) if asyncio.iscoroutinefunction(func) else func(*args, **kwargs)
+            duration = time.time() - start_time
+            self.metrics.record_request(agent_name, duration, success=True)
+            logger.info("Request processed successfully", 
+                       agent=agent_name, 
+                       duration_ms=duration*1000,
+                       success=True)
+            return result
+        except Exception as e:
+            duration = time.time() - start_time
+            success = False
```

Ruff (0.12.2)

Local variable `success` is assigned to but never used

Remove assignment to unused variable `success`

(F841)